@here/harp-mapview/lib/DataSource.ts

```diff
@@ -151,6 +151,14 @@ export abstract class DataSource extends THREE.EventDispatcher {
         // to be overloaded by subclasses
     }
 
+    /**
+     * Clears this `DataSource`. Clearing as opposed to [[dispose]] just resets any internal state,
+     * as opposed to actually destroying allocated objects.
+     */
+    clear() {
```

Comment:
Continuing internal discussion.

> it acceptable to add a clear method to a DataSource

For sure it's acceptable, but naming could be better like `resetState`, `clearCaches` ...

On the other hand, the responsibility for this cache is in `TileDataSource` so this class should know about this and just clear any needed cache in it's `.setStyleSet` implementation and we don't need `DataSource#clear` (and discussion about it's name and semantics).

Reply:
Agreed.